backend/associations/views/library.py

```diff
@@ -44,13 +49,71 @@ def get_serializer_class(self):
         return LibrarySerializer
 
 
+class LoanableFilter(FilterSet):
+    status = MultipleChoiceFilter(
+        choices=(
+            ("AVAILABLE", "AVAILABLE"),
+            ("BORROWED", "BORROWED"),
+            ("REQUESTED", "REQUESTED"),
+        ),
+        method="filter_status",
+    )
+
+    class Meta:
+        model = Loanable
+        fields = ("status", "library")
+
+    def filter_status(self, queryset, _, filter):
+        """Filter by AVAILABLE, BORROWED or REQUESTED.
+                                   | The loanable is available | The loanable is borrowed |
+        There are PENDING loans    | REQUESTED                 | Impossible               |
+        There are no PENDING loans | AVAILABLE                 | BORROWED                 |
+        """
+
+        # No filter or every filter.
+        if len(filter) == 0 or len(filter) == 3:
+            return queryset
+
+        ids = set()
+
+        if "AVAILABLE" in filter:
+            ids.update(
+                [
+                    x.id
+                    for x in queryset
+                    if x.is_available() and x.number_of_pending_loans == 0
+                ]
+            )
```

Comment:
Replace with additional filters

Reply:
This filter relies on a property is_available, so it cannot be filtered in a Queryset.filter

Reply:
It can,  it's just not built in for now